pandas/tests/indexes/common.py

```diff
@@ -721,3 +721,22 @@ def test_fillna(self):
                 expected[1] = True
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
+
+    def test_is_index_empty(self):
+        """ GH 13207 Bug Fix """
+        for ind in self.indices.values():
+            if ind.size == 0:
```

Comment:
this only tests the non-empty part, you need to construct a new index of the same type just with no size and test for empty.